src/main/java/hudson/plugins/mercurial/MercurialSCM.java

```diff
@@ -765,6 +770,9 @@ private void clone(Run<?, ?> build, Launcher launcher, FilePath repository, Node
                 args.add("share");
                 args.add("--noupdate");
                 args.add(cachedSource.getRepoLocation());
+                if (new VersionNumber(hg.version()).compareTo(new VersionNumber("3.3")) >= 0) {
+                    args.add("-B");
```

Comment:
Long form of the option is preferred: makes it easier to understand what is going on from the source code.